caikit/runtime/service_generation/rpcs.py

```diff
@@ -95,12 +96,22 @@ def create_request_data_model(self, package_name: str) -> Type[DataBase]:
 
     def create_rpc_json(self, package_name: str) -> Dict:
         """Return json snippet for the service definition of this RPC"""
-        output_type_name = self.return_type.get_proto_class().DESCRIPTOR.full_name
+        if isiterable(self.return_type):
```

Comment:
Looks like you're using this again for type analysis rather than instance analysis. I don't think we actually used this at all in `caikit` proper, but some of our users might have been using to check instances, so we should make sure it wouldn't be better to do something like `isiterable_type` as a separate check.

Reply:
Yeah, I could be convinced to split that into a separate function